src/librustc_driver/driver.rs

```diff
@@ -351,6 +352,14 @@ pub fn compile_input(
         sess.print_perf_stats();
     }
```

Comment:
one question I'm wondering: how do we find the "total compilation time" number? I'd like to see some "start activity" and "end activity" that takes place at a really high-up place, like `main`, that should truly account for *everything*

Reply:
I believe this is just a matter of creating the `SelfProfiler` earlier in the process lifecycle. Right now it's created when `Session::new()` is called, but there's no real reason it couldn't be created earlier.

Reply:
That's probably early enough -- when does it make its final measurement?

Reply:
At the end of `compile_input()` right after we run the linker and output the other performance diagnostics.